### PR TITLE
New Click Command

### PR DESCRIPTION
**Pull request by adammentzer**
Added Click command to display the number of projects. Options exist to specify for projects within the bounds of the specified number of bills (inclusive) and/or less than x days old.

### CODE REVIEW

File: ihatemoney/manage.py [hunk 1]
```diff
@@ -94,5 +95,97 @@ def delete_project(project_name):
...
+):
+    """Displays the number of projects. Options exist to specify for \
+        projects within the bounds of the specified number of bills (inclusive) and/or less than x days old. """
+    if bills_low and bills_high:
+
+        # Make sure low < high
+        if bills_high < bills_low:
+            temp = bills_high
+            bills_high = bills_low
+            bills_low = temp
```
Comment (Glandos) [inline ihatemoney/manage.py:137]:
```suggestion
        if bills_high < bills_low:
            (bills_low, bill_high) = bills_high, bills_low
```

File: ihatemoney/manage.py [hunk 2]
```diff
@@ -94,5 +95,97 @@ def delete_project(project_name):
...
+
+        if num_days:
+            projects = [
+                pr
+                for pr in Project.query.all()
+                if pr.get_bills().count() >= bills_low
+                and pr.get_bills().count() <= bills_high
```
Comment (Glandos) [inline ihatemoney/manage.py:144]:
```suggestion
                if  bills_low <= pr.get_bills().count() <= bills_high
```

File: ihatemoney/manage.py [hunk 3]
```diff
@@ -94,5 +95,97 @@ def delete_project(project_name):
...
+        else:
+            projects = Project.query.all()
+
+        click.echo(len(projects))
+        if emails:
+            list_emails = ", ".join(set([pr.contact_email for pr in projects]))
+            click.echo(list_emails)
+        if names:
+            proj_names = [pr.name for pr in projects]
+            click.echo(proj_names)
```
Comment (Glandos) [inline ihatemoney/manage.py:187]: This is duplicated code from line 155-161.

**Review by Glandos — CHANGES_REQUESTED** [review]
I've added some small Pythonic changes.

But globally, it would better to have a list of predicates to filter the list of project, and execute this list comprehension `project = []` only once.
I bet this is possible with SQLAlchemy, but don't have the time to find it, so, it will be fine for me with Python lambdas :)

**Comment by Glandos** [comment]
Also, maybe a default value of 0 for `bills_low` should be a better default, instead of having to chose both low and high.

**Comment by almet** [comment]
Thanks, this has been done in #1115

**Pull request closed without merging**